client/tests/integration/triggers/data_trigger.rs

```diff
@@ -20,16 +20,10 @@ fn must_execute_both_triggers() -> Result<()> {
             [instruction.clone()],
             Repeats::Indefinitely,
             account_id.clone(),
-            // FIXME: rewrite the filters using the builder DSL https://github.com/hyperledger/iroha/issues/3068
-            TriggeringFilterBox::Data(BySome(DataEntityFilter::ByDomain(BySome(
-                DomainFilter::new(
-                    AcceptAll,
-                    BySome(DomainEventFilter::ByAccount(BySome(AccountFilter::new(
-                        AcceptAll,
-                        BySome(AccountEventFilter::ByCreated),
-                    )))),
-                ),
-            )))),
+            TriggeringEventFilterBox::Data(DataEventFilter::ByAccount(AccountEventFilter {
+                id_matcher: None,
+                event_matcher: Some(AccountEventMatcher::ByCreated),
+            })),
```

Comment:
in all these cases you should make constructors so it would look something like this:

```suggestion
            let account_event_filter = AccountEventFilter::new(AccountEventMatcher::ByCreated);
            DataEventFilter::ByAccount(account_event_filter),
```

it is only an unfortunate consequence of the fact that we have `iroha_core` a dependency of `test_network` (which then activates `transparent_api`) that we can use internal struct fields in client tests. Avoid doing it because we will forbid this in the future

Reply:
So, you want to prohibit creation of `*EventFilter` structs via struct literals?

Why though? Unlike many other types, event filters don't really have any invariants, all of them are valid by-construction..

Reply:
> So, you want to prohibit creation of *EventFilter structs via struct literals?

it is prohibited by the `model` macro. That macro hides internal fields of all structures. The fact that you can access these fields in `iroha_client/tests` is only coincidental because it has `iroha_core` as `dev-dependency`. Since we want tests to be representative of how you would actually use our code, then you can't use internal fields

Reply:
We're doing this so that we can, in the future, substitute `getters`/`setters` with FFI calls. It gives us the flexibility to replace structures with opaque pointers

Reply:
I see, I'll make the constructors then

Reply:
Did builder-like API for constructors and also derived getters to introspect the filters